src/sinks/opendal_common.rs

```diff
@@ -0,0 +1,218 @@
+//! opendal_common provide real sink supports for all opendal based services.
+//!
+//! # TODO
+//!
+//! opendal service now only support very basic sink features. To make it
+//! useful, we need to add the following features:
+//!
+//! - Batch events
+//! - Error handling
```

Comment:
**issue (blocking)**: I think rudimentary error handling and batching should be spiked out before merging.

We'd want to make sure end-to-end acknowledgements are working properly, batching is less required - but good to see if the implementation isn't too large.